sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -405,6 +417,141 @@ Exceptions:
+affect any executable graphs this node is part of until it is passed to the
+executable graph's update function.
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+Parameters:
+
+* `executionRange` - The new value for the ND-Range.
+
+Exceptions:
+
+* Throws with error code `invalid` if `Dimensions` does not match the dimensions
+  of the nd_range the kernel was originally created with.
+
+* Throws with error code `invalid` if the kernel node was originally created
+  with a `sycl::range`.
+
+* Throws with error code `invalid` if the type of the node is not a kernel
+  execution.
+
+|
+[source,c++]
+----
+template <int Dimensions>
+void update_range(range<Dimensions> executionRange);
+----
+| Updates the execution Range for this node with a new value. This new value
+will not affect any executable graphs this node is part of until it is
+passed to the executable graph's update function.
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+Parameters:
+
+* `executionRange` - The new value for the Range.
+
+Exceptions:
+
+* Throws with error code `invalid` if `Dimensions` does not match the dimensions
+  of the range the kernel was originally created with.
+
+* Throws with error code `invalid` if the kernel node was originally created
+  with a `sycl::nd_range`.
+
+* Throws with error code `invalid` if the type of the node is not a kernel
+  execution.
+
+|===
+
+==== Dynamic Parameters
+
+[source,c++]
+----
+namespace ext::oneapi::experimental{
+template <typename ValueT>
+class dynamic_parameter {
+public:
+  dynamic_parameter(command_graph<graph_state::modifiable> graph, const ValueT &initialValue);
+
+  void update(const ValueT& newValue);
+};
+}
+----
+
+Dynamic parameters are arguments to a node's command-group which can be updated
+by the user after the node has been added to a graph. Updating the value of a
+dynamic parameter will be reflected in the modifiable graph which contains this
+node. These updated nodes can then be passed to an executable graph to update
+it with new values.
+
+The type of the underlying object a dynamic parameter represents is set at
+compile time using a template parameter. This underlying type can be an
+accessor, a pointer to a USM allocation, scalar passed by value, or a raw byte
+representation of the argument. The raw byte representation is intended to
+enable updating arguments set using
+link:../proposed/sycl_ext_oneapi_raw_kernel_arg.asciidoc[sycl_ext_oneapi_raw_kernel_arg].
+
+Dynamic parameters are registered with nodes in a modifiable graph, with each
+registration associating one or more node arguments to the dynamic parameter
+instance. Registration happens inside the command-group that the node represents
+, and is done when dynamic parameters are set as parameters to the kernel using
+`handler::set_arg()`/`handler::set_args()`). It is valid for a node argument to
```

Comment:
```suggestion
instance. Registration happens inside the command-group that the node represents,
and is done when dynamic parameters are set as parameters to the kernel using
`handler::set_arg()`/`handler::set_args()`. It is valid for a node argument to
```

Reply:
I'll remove the extra `)` but moving that comma would put that line over 80 columns, although it might be slightly more readable I'd prefer to keep it to 80 columns for easier automatic formatting probably.

Reply:
the newline before the comma adds whitespace to the rendered view after "represents" and before the comma, so if
the comma takes the line over 80 chars then I think `represents,` should be moved to the next line.

Reply:
That is fair I missed that part. I had also thought that change would interfere with the automatic text wrapping I'm doing but it doesn't actually so that's fine, I'll make that change.